diff --git a/CMakeLists.txt b/CMakeLists.txt
index ec6ee5c..5b74b02 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,12 @@ CMAKE_MINIMUM_REQUIRED(VERSION 3.10.0)
 # Project name and version
 PROJECT(libcaer VERSION 3.3.12 LANGUAGES C CXX)
 
+add_definitions(-D_GNU_SOURCE)
+add_definitions(-D_BSD_SOURCE)
+add_definitions(-D_DEFAULT_SOURCE)
+add_definitions(-D__USE_BSD)
+add_definitions(-D_USE_BSD)
+
 # Basic setup, useful variables, see docs.
 SET(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmakemod ${CMAKE_MODULE_PATH})
 INCLUDE(inivation-setup)
diff --git a/src/davis_rpi.c b/src/davis_rpi.c
index 02d9938..53170ea 100644
--- a/src/davis_rpi.c
+++ b/src/davis_rpi.c
@@ -4,6 +4,7 @@
 #include <linux/spi/spidev.h>
 #include <linux/types.h>
 #include <sys/ioctl.h>
+#include <linux/mman.h>
 #include <sys/mman.h>
 #include <unistd.h>
 
diff --git a/src/ringbuffer.c b/src/ringbuffer.c
index 4bd8bbd..3e3a3f1 100644
--- a/src/ringbuffer.c
+++ b/src/ringbuffer.c
@@ -2,7 +2,10 @@
 
 #include "portable_aligned_alloc.h"
 
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+#else
 #include <stdalign.h> // To get alignas() macro.
+#endif
 #include <stdatomic.h>
 
 // Alignment specification support (with defines for cache line alignment).
